### PR TITLE
improve sparql-qa-chain

### PR DESCRIPTION
**Pull request by felixocker**
**Description:**
* fixed the SPARQL queries for schema extraction - in some cases not all classes were loaded
* improved the prompt for creating natural language answers from the query results
* fixed the RDF example - the content was not retrieved from the graph DB due to the name not matching the one in the graph
* added an OWL example with a local file - simplified, but inspired by @LorenzBuehmann's example in #8907

**Issue:**
loosely related to issue #8907 and the discussion within PR #11240. the issues mentioned there were resolved in #9136, but made me notice another issue where classes were missing in the extracted schema

**Dependencies:**
none

**Tag maintainer:**
@baskaryan

### CODE REVIEW

File: docs/docs/use_cases/graph/graph_sparql_qa.ipynb
Comment (baskaryan) [inline docs/docs/use_cases/graph/graph_sparql_qa.ipynb:1]: could we add to the docs an example of customizing the different prompts? imagine no prompt will be perfect for every use case so it's important to show how to customize

**Comment by LorenzBuehmann** [comment]
Hi,

I can see two limitations resp. opportunities for improvement here:

1. using the local name of an IRI is not the intended way of providing a human readable name in RDF. The best practice is usually to use `rdfs:label` or similar properties (e.g. skos:prefLabel, skos:altLabel, ...). For example, in Wikidata all IRIs just have IDs like `https://www.wikidata.org/entity/Q5`, same holds for many biomedical domain ontologies.
2. we should add a config option to restrict the text to a particular language, e.g. to English, Spanish, etc. - even if the LLM can handle different languages, in my opinion it'S way cleaner to have just one language per prompt. And in RDF you can indeed have `rdfs:comment` literals in multiple languages. The current code would therefore return multiple rows for the same class or property with descriptions in different languages. This would also make the prompt larger than necessary. 

I hope my comments make sense to some extent.

File: docs/docs/use_cases/graph/graph_sparql_qa.ipynb
Comment (felixocker) [reply docs/docs/use_cases/graph/graph_sparql_qa.ipynb:1]: yeah, sounds good. I added a section `customization` to the jupyter notebook in fd0ba32

**Comment by felixocker** [comment]
@LorenzBuehmann thanks for your input, I fully agree.
I added support for labels in a062f4d and ad187a4 allows setting a `language` for labels and comments (default is no language)
Note that if a language is set, labels and comments should actually be available in that language for all notions, otherwise they may be ignored. But this seems like a reasonable assumption to me, at least for now.

**Comment by felixocker** [comment]
@baskaryan could you have a look at this? I think it's ready to merge, but any feedback is appreciated.

**Pull request closed without merging**